### PR TITLE
Feature cucumber test

### PR DESCRIPTION
**Pull request by DVA09**

### CODE REVIEW

File: BDD/features/login_page.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Feature: Test Login page
+
```
Comment (Alex19777) [inline BDD/features/login_page.feature]: Add more tests and use background step for common steps

File: test/pages/ProductPage.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    get sortOption3() {
+        return $("//*[@id='header_container']/div[2]/div/span/select/option[3]");
+    }
+
+    get sortOption4() {
+        return $("//*[@id='header_container']/div[2]/div/span/select/option[4]");
+    }
+
```
Comment (Alex19777) [inline test/pages/ProductPage.js:69]: Line

File: test/pages/ProductPage.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    //Footer locators
+    get social() {
+        return $$(".social");
+    }
+
+    get copyText() {
+        return $$("//*[@id='page_wrapper']/footer/div");
+    }
+
```
Comment (Alex19777) [inline test/pages/ProductPage.js:101]: Line

File: test/specs/product_page.spec.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    });
+
+    it("Burger Menu item4 test", async () => {
+
+        await expect(ProductPage.iconBurgerMenuItem4).toHaveText('Reset App State');
+
+        await expect(ProductPage.iconBurgerMenuItem4).toBeClickable();
+
+        await ProductPage.iconBurgerMenuCross.click();
+
```
Comment (Alex19777) [inline test/specs/product_page.spec.js:71]: Here and in all same places

File: test/utils/product.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+    "url": "https://www.saucedemo.com/inventory.html"    
+
```
Comment (Alex19777) [inline test/utils/product.json]: same

File: test/pages/ProductPage.js (hunk 2 above)
Comment (DVA09) [reply test/pages/ProductPage.js:69]: Fixed it

File: test/pages/ProductPage.js (hunk 3 above)
Comment (DVA09) [reply test/pages/ProductPage.js:101]: Fixed it

File: BDD/features/login.page.feature [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    Then burger menu Item4 has text Reset App State
+    Then item4 is clickable
+    When close burger menu
+    Then product page title has text
+    Then basket icon to be displayed on the page
+    Then basket icon is clickable
+    Then secondary header has text
+    Then filter icon to be displayed on the page
+    Then filter icon is clickable
+    Then filter has 4 options
```
Comment (Alex19777) [inline BDD/features/login.page.feature]: Use And here

File: BDD/features/login.page.feature [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    When open list with filter options 4
+    When click on 4th filter option
+    Then 4th filter option to be displayed
+    Then 4th filter option has text
+    Then names of the products is clickable
+    Then images of the products is clickable
+    Then button of the products is clickable
+    Then names of the products don't have brackets
+    Then description of the products don't have brackets
+    Then social icons to be clickable
```
Comment (Alex19777) [inline BDD/features/login.page.feature]: Some checks you can combine into one step

File: BDD/features/login_page.feature (hunk 1 above)
Comment (DVA09) [reply BDD/features/login_page.feature]: Done it

File: test/specs/product_page.spec.js (hunk 4 above)
Comment (DVA09) [reply test/specs/product_page.spec.js:71]: Deleted extra empty lines

File: test/utils/product.json (hunk 5 above)
Comment (DVA09) [reply test/utils/product.json]: Fixed it

File: BDD/features/login.page.feature (hunk 6 above)
Comment (DVA09) [reply BDD/features/login.page.feature]: Add And to the Scenario steps

File: BDD/features/login.page.feature (hunk 7 above)
Comment (DVA09) [reply BDD/features/login.page.feature]: Combined steps

**Review by Alex19777 — APPROVED** [review]

**Pull request merged**